acceptance/tests/options/config_file/ttls_cache_system_uptime.rb

```diff
@@ -56,39 +68,39 @@
     #   assert_equal(expected, cache_content)
     # end
 
-    step "calling facter without a query will cache the entire group" do
+    step 'calling facter without a query will cache the entire group' do
       _output = on(agent, facter)
 
       stdout = agent.cat("#{cache_folder}/uptime")
       cache_content = JSON.parse(stdout)
-      ["system_uptime.days",
-        "uptime_days",
-        "system_uptime.hours",
-        "uptime_hours",
-        "system_uptime.seconds",
-        "uptime_seconds",
-        "system_uptime.uptime",
-        "uptime",
-        "cache_format_version"].each do |key|
+      ['system_uptime.days',
+        'uptime_days',
+        'system_uptime.hours',
+        'uptime_hours',
+        'system_uptime.seconds',
+        'uptime_seconds',
+        'system_uptime.uptime',
+        'uptime',
+        'cache_format_version'].each do |key|
 
           assert_equal(true, cache_content.has_key?(key))
         end
     end
 
-    step "calling a single fact fron the cached group will not overwrite the file" do
+    step 'calling a single fact fron the cached group will not overwrite the file' do
       _output = on(agent, facter('system_uptime.seconds'))
 
       stdout = agent.cat("#{cache_folder}/uptime")
       cache_content = JSON.parse(stdout)
-      ["system_uptime.days",
-        "uptime_days",
-        "system_uptime.hours",
-        "uptime_hours",
-        "system_uptime.seconds",
-        "uptime_seconds",
-        "system_uptime.uptime",
-        "uptime",
-        "cache_format_version"].each do |key|
+      ['system_uptime.days',
+        'uptime_days',
+        'system_uptime.hours',
+        'uptime_hours',
+        'system_uptime.seconds',
+        'uptime_seconds',
+        'system_uptime.uptime',
+        'uptime',
+        'cache_format_version'].each do |key|
 
           assert_equal(true, cache_content.has_key?(key))
```

Comment:
You should check that the facter returned value for uptime_seconds coincides with the one found in the cache file. Thus we make sure that facter doesn't solve the cached facts. Also you need to check that the cache file hasn't been rewritten. agent.ls_ld should help